templates/vue-demo-store/layouts/default.vue

```diff
@@ -4,6 +4,11 @@ export default {
 };
 </script>
 <script setup lang="ts">
+// in development mode we test if your API connection is working
+if (process.env.NODE_ENV == "development") {
```

Comment:
this won't work in SSG (with settings in nuxt.config.ts -> `ssr: false`). 

we can move https://github.com/shopware/frontends/pull/404/files#diff-c0db108b331ef293b83598ea004b65e640ea63cbe550676d7c18248c2b60f8d2R20-R32 into `error.vue` file in order to add some `console.error` instead of invoking a getter for session context in a composable. Then the `useConnectionTest` can be removed. 

Pros: 
- will work on ssg as well as ssr
- one file less 👍🏼 

Cons:
- it does not follow an "interceptor" approach (but it's ok as we plan to migrate to new api client)